src/python/pants/option/options.py

```diff
@@ -273,6 +273,22 @@ def verify_configs(self, global_config: Config) -> None:
             )
         global_config.verify(section_to_valid_options)
 
+    def verify_args(self):
+        # Consume all known args, and see if any are left.
+        # This will have the side-effect of precomputing (and memoizing) options for all scopes.
+        for scope in self.known_scope_to_info:
+            # Currently the legacy parser raises UnknownFlagsError in for_scope().
+            # When we switch to the native parser we will manually check get_unconsumed_flags().
+            self.for_scope(scope)
+        # TODO: Uncomment this when we switch to the native parser.
```

Comment:
Hm, potential point for debate: this is arguably part of the interface of the parser, and thus potentially we might want to know if the legacy and native parsers give different results? As in, do we want to be running them in parallel?

Reply:
They both throw exceptions, so doing so in parallel would require some changes to the currently very stable `for_scope()` method, which I would much rather not modify at this late stage... But the tests cover both in practice.